File: droidwarn.py
import sys, time

# Yes, this code has absolutely nothing to do with Android.
# However, Google just announced that starting in 2026/2027, 
# they will force all Android app developers to submit personal
# identity details directly to them—even for apps outside the 
# Play Store. This destroys developer privacy and independent
# ecosystems like F-Droid.
# 
# Learn more on https://ricktree.free.nf/keepandroidopen and https://keepandroidopen.org

def droidwarn(interactive:bool=True, dependency:bool=False):
    # Use ANSI escape codes for subtle color coding (e.g., Yellow/Red)
    YELLOW = "\033[93m"
    BOLD = "\033[1m"
    END = "\033[0m"
    
    msg = (
        f"\n{YELLOW}{BOLD}🚨 Tech Freedom Alert:{END} Google is centralizing Android "
        "app distribution via identity verification starting late 2026, threatening "
        "open-source ecosystems like F-Droid. Even if this project isn't an Android app, "
        f"corporate gatekeeping affects all software freedom.\n"
        f"👉 Learn more & take action: {BOLD}https://keepandroidopen.org{END}\n"
    )
    print(msg, file=sys.stderr)
    if interactive:
        confirmation = input("\nDo you understand? (type \"Yes, I understand, Keep Android Open!\" to respond \"Yes\", any other answer is a \"No\")\n>>> ")
        if confirmation == "Yes, I understand, Keep Android Open!":
            pass
        else:
            print("Please understand to the warning and try again, we do not allow people to run this without understanding it.")
            time.sleep(3)
            exit()
    elif dependency:
        print("Since this is a dependency, we won't give you any friction using this dependency/module.")

# Quick usage, ideal for pygame (must use before doing "import pygame")
droidwarn()

# Implement this on your module as well!
# Usage:
# droidwarn(False, True)

File: droidwarn.java
/** 

I mainly use this to target command line based projects
You are free to make it GUI based manually

*/

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class droidwarn {
    public static void droidwarn(boolean interactive, boolean dependency) {
        String yellow = "\033[93m";
        String bold = "\033[1m";
        String end = "\033[0m";
        String confirmationPhrase = "Yes, I understand, Keep Android Open!";

        String msg = "\n" + yellow + bold + "🚨 Tech Freedom Alert:" + end + " Google is centralizing Android " +
                "app distribution via identity verification starting late 2026, threatening " +
                "open-source ecosystems like F-Droid. Even if this project isn't an Android app, " +
                "corporate gatekeeping affects all software freedom.\n" +
                "👉 Learn more & take action: " + bold + "https://keepandroidopen.org" + end + "\n";

        System.err.print(msg);

        if (interactive) {
            System.out.println("\nDo you understand? (type \"" + confirmationPhrase + "\" to respond \"Yes\", any other answer is a \"No\")");
            System.out.print(">>> ");
            try {
                BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
                String input = reader.readLine();
                if (input != null && input.trim().equals(confirmationPhrase)) {
                    return;
                } else {
                    System.err.println("Please understand the warning and try again, we do not allow people to run this without understanding it.");
                    Thread.sleep(3000);
                    System.exit(1);
                }
            } catch (Exception e) {
                System.exit(1);
            }
        } else if (dependency) {
            System.err.println("Since this is a dependency, we won't give you any friction using this dependency/module.");
        }
    }

    public static void main(String[] args) {
        droidwarn(true, false);
    }
}
